digits/extensions/data/imageProcessing/data.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (c) 2016, NVIDIA CORPORATION.  All rights reserved.
+from __future__ import absolute_import
+
+import math
+import os
+import random
+
+import numpy as np
+
+from digits.utils import image, subclass, override, constants
+from ..interface import DataIngestionInterface
+from .forms import DatasetForm
+
+TEMPLATE = "template.html"
+
+
+@subclass
+class DataIngestion(DataIngestionInterface):
+    """
+    A data ingestion extension for an image processing dataset
+    """
+
+    def __init__(self, **kwargs):
+        """
+        the parent __init__ method automatically populates this
+        instance with attributes from the form
+        """
+        super(DataIngestion, self).__init__(**kwargs)
+
+        self.random_indices = None
+
+        if not 'seed' in self.userdata:
+            # choose random seed and add to userdata so it gets persisted
+            self.userdata['seed'] = random.randint(0, 1000)
+
+        random.seed(self.userdata['seed'])
```

Comment:
Good usage of the user seed. This reminds me however of the way shuffling, ofc an random op, is done elsewhere in DIGITS:
In object detection datasets: https://github.com/NVIDIA/DIGITS/blob/master/digits/extensions/data/objectDetection/data.py#L235
and actually also in folder parsing
https://github.com/NVIDIA/DIGITS/blob/master/tools/parse_folder.py#L416
It's not clear to me if that seed from folder parsing is random or if the scope is persistent from somewhere else. In any case, in uploading a folder it might also be nice to be able to randomize or fix the seed, during shuffling in folder parsing.